javascript/javascript1/week4/voice_assistant.js

```diff
@@ -1,68 +1,81 @@
+  datePattern: /What day is it today\?/i,
+  timerPattern: /Set a timer for (\d+) min/i,
+};
+const commands = {
+  [patterns.namePattern]: {
+    reply: (input) => {
+      const userName = getUserName(input);
+      if (userName) {
+        state.userName = userName;
+        return `Nice to meet you, ${state.userName}`;
+      } else {
+        return "Sorry, I could not determine your name.";
+      }
+    },
+  },
+  [patterns.addPattern]: {
+    reply: (input) => {
+      return addToDo(input);
+    },
+  },
+  [patterns.removePattern]: {
+    reply: (input) => {
+      return removeToDo(input);
+    },
+  },
+  [patterns.getNamePattern]: {
+    reply: () => {
+      if (state.userName) {
+        return `Your name is ${state.userName}.`;
+      } else return `Sorry, I cannot determine your name.`;
+    },
+  },
+  [patterns.datePattern]: {
+    reply: () => {
+      return currentDate();
+    },
+  },
+  [patterns.mathPattern]: {
+    reply: (input) => calculateResult(input),
+  },
+
+  [patterns.tempPattern]: {
+    reply: (input) => smartWear(input),
+  },
+  [patterns.timerPattern]: {
+    reply: (input) => setTimer(input),
+  },
+  [patterns.showToDoPattern]: {
+    reply: () => state.toDoList,
+  },
+};
 
-// Tried to create an object instead of if else statements in getReply() function
-// but it was challenging to keep input somewhat dynamic.
-// const commands = {
-//   "Hello, my name is [name]": (input) => {
-//     userName = getUserName(input);
-//     return `Nice to meet you, ${userName}`;
-//   },
-//   "What is my name?": () => `Your name is ${userName}`,
-//   "Add [todo] to my to do list.": (input) => {
-//     const toDo = addToDo(input);
-//     return `${toDo} added to your to do list.`;
-//   },
-// };
-let userName = null;
-const toDoList = [];
+//main function
 function getReply(input) {
-  const addPattern = /Add (.+) to my to do list\./i;
-  const removePattern = /Remove (.+) from my to do list\./i;
-  const mathPattern = /what is (\d+)\s*([+\-*\/])\s*(\d+)\?/i;
-  if (input === "Hello, my name is Tanya") {
-    userName = getUserName(input);
-    return `Nice to meet you, ${userName}`;
-  } else if (input === "What is my name?") {
-    return `Your name is ${userName}`;
-  } else if (input === "How are you today?") {
-    return "Good!";
-  } else if (addPattern.test(input)) {
-    const toDo = addToDo(input);
-    return `${toDo} added to your to do list.`;
-  } else if (removePattern.test(input)) {
-    return removeToDo(input);
-  } else if (input === "What is my to do?") {
-    return `You have ${toDoList.length} items in your to do list: ${toDoList}`;
-  } else if (input === "What day is it today?") {
-    return currentDate();
-  } else if (mathPattern.test(input)) {
-    return calculateResult(input);
+  for (const pattern in patterns) {
+    if (patterns[pattern].test(input)) {
+      const replyFunction = commands[patterns[pattern]].reply;
+      return replyFunction(input);
+    }
```

Comment:
Very minimal and sleek solution which you can easily expand upon with new rules 👏

Reply:
Thank you!